src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.dao.interfaces.MovieDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```

Comment:
you may move session factory to class variables, since it repeats twice

Reply:
done